src/libcore/heap.rs

```diff
@@ -125,15 +125,21 @@ impl Layout {
     /// Constructs a `Layout` suitable for holding a value of type `T`.
     pub fn new<T>() -> Self {
         let (size, align) = size_align::<T>();
-        Layout::from_size_align(size, align).unwrap()
+        debug_assert!(Layout::from_size_align(size, align).is_some());
```

Comment:
Could we add a comment here as to why we have this unsafe code? Otherwise I worry someone will remove it down the line since it feels unnecessary.

Either way, I feel like it's unfortunate we have to do this :/